02-deploying-an-app/006-cleaning-up.md

```diff
@@ -0,0 +1,52 @@
+# Cleaning up
+
+When you have finished working with your initial deployment, please clean up the resources you have created. This helps to keep the cloud platform repositories well-organised, and speeds up deployments and changes to the cluster (because the build process doesn't have to spend time managing unnecessary resources). It also helps to keep our hosting costs down.
+
+The resources to be removed are:
+
+* The ECR which stores your docker images
+* Your namespace in the cluster. This contains all of the pods, containers and other cluster resources for your application. Removing the cluster namespace will automatically clean up all of its contents.
+
+## Removing resources from your namespace
+
+Do this if you want to empty your namespace, e.g. to reuse it for another section of the user guide, or to experiment for yourself. You do not need to do this if you are removing the whole namespace - any contents will be deleted automatically, in that case.
+
+Use the `kubectl` command to remove resources from your namespace, e.g.
+
+      kubectl delete deployment helloworld-rubyapp --namespace [your namespace]
+      kubectl delete service rubyapp-service --namespace [your namespace]
+      kubectl delete ingress helloworld-rubyapp-ingress --namespace [your namespace]
+
+Use `kubectl get ...` to determine the correct names, and to confirm deletion.
+
+You can find more information about `kubectl` [here][kubectl].
+
+## Removing your ECR
+
+Your [ECR][ecr] was created by adding an `ecr.tf` file to the Cloud Platform [environments repository][envrepo].
+
+To delete your ECR, once you no longer need it, requires two steps:
+
+1. Remove the `ecr.tf` file from the [environments repository][envrepo]. This prevents the ECR from being automatically recreated the next time the cluster configuration is applied.
+1. A cluster administrator needs to manually delete the ECR (either via the Amazon AWS web console, or using the [AWS CLI][awscli].
+
+## Removing your namespace
+
+Namespaces are created by adding YAML config files to the Cloud Platform [environments repository][envrepo].
+
+To delete a namespace requires two steps:
+
+1. Remove the YAML config files from the [environments repository][envrepo]. This prevents the namespace from being automatically recreated the next time the cluster configuration is applied.
+1. A cluster administrator needs to run a `kubectl delete namespace [name of the namespace]` command to delete the namespace and its contents.
+
+## Raising a pull request
+
+When you have finished with your sandbox namespace, and its associated ECR, please create a fork of the [environments repository][envrepo] with your environment's sub-directory of the `namespaces/cloud-platform-live-0.k8s.integration.dsd.io` directory removed, and raise a [pull request][pr] to merge your fork into master. Once merged, this will remove the YAML files which define your namespace, and also the `ecr.tf` file which defines your ECR.
```

Comment:
We typically give write access to the repository so that people can branch off instead of forking. Do you think we should adjust wording? Forking would just work, however, assuming no special permissions.

Reply:
You're absolutely right - I should have talked about branching, not forking. I'll change it.